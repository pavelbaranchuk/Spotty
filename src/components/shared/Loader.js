import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";

const styles = theme => ({
  progress: {
    color: theme.palette.primary.main
  },
  progressContainer: {
    width: "10rem",
    height: "10rem",
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)"
  }
});

function Loader(props) {
  const { classes } = props;
  return (
    <div className={classes.progressContainer}>
      <CircularProgress size={160} className={classes.progress} />
    </div>
  );
}

Loader.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles, { withTheme: true })(Loader);
