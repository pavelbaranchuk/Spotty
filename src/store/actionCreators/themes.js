import * as actionTypes from "../actionTypes";
import { ThemeService } from "../../services/ThemeService";
import { AuthService } from "../../services/AuthService";

export function setUserTheme(themeType) {
  return async dispatch => {
    try {
      dispatch({ type: actionTypes.SET_USER_THEME_START });
      const user = await AuthService.check();
      await ThemeService.setTheme(user.uid, themeType);
      dispatch({
        type: actionTypes.SET_USER_THEME_SUCCESS,
        themeType
      });
    } catch (error) {
      dispatch({ type: actionTypes.SET_USER_THEME_FAIL });
    }
  };
}

export function setDefaultTheme() {
  return {
    type: actionTypes.SET_DEFAULT_THEME
  };
}
