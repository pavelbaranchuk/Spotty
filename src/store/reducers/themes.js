import * as actionTypes from "../actionTypes";

export const initialState = {
  themeType: "light"
};

export default function themeReducer(state = initialState, action) {
  switch (action.type) {
    case actionTypes.SET_USER_THEME_SUCCESS:
      return {
        themeType: action.themeType
      };
    case actionTypes.FETCH_USER_AND_THEME_SUCCESS:
    case actionTypes.SIGN_IN_SUCCESS:
      return {
        themeType: action.themeType
      };
    case actionTypes.FETCH_USER_AND_THEME_FAIL:
    case actionTypes.SET_DEFAULT_THEME:
      return {
        themeType: initialState.themeType
      };
    default:
      return state;
  }
}
