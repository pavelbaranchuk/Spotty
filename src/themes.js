import { createMuiTheme } from "@material-ui/core/styles";
import "typeface-roboto";

export const lightTheme = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  palette: {
    type: "light",
    primary: {
      main: "#1e88e5"
    },
    disable: "#cccccc"
  },
  props: {
    drawer: {
      drawerWidth: 200
    },
    icons: {
      iconColor: "#0000008a"
    },
    mediaPlayer: {
      mediaPlayerHeight: "6.8rem",
      mediaPlayerBackgroundColor: "#fefefe",
      mediaPlayerPlayButtonBackground: "#1e88e5"
    },
    appBar: {
      appBarHeight: "4rem",
      appBarBackgroundColor: "#1e88e5"
    }
  }
});

export const darkTheme = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  palette: {
    type: "dark",
    primary: {
      main: "#1e88e5"
    },
    disable: "#cccccc"
  },
  props: {
    drawer: {
      drawerWidth: 200
    },
    mediaPlayer: {
      mediaPlayerHeight: "6.8rem",
      mediaPlayerBackgroundColor: "#424242",
      mediaPlayerPlayButtonBackground: "#838383"
    },
    icons: {
      iconColor: "#fff"
    },
    appBar: {
      appBarHeight: "4rem",
      appBarBackgroundColor: "#424242"
    }
  }
});
