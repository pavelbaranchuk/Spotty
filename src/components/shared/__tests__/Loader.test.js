import React from "react";
import renderer from "react-test-renderer";
import Loader from "../Loader";

import { MuiThemeProvider } from "@material-ui/core/styles";
import { lightTheme } from "../../../themes";

jest.mock("@material-ui/core/CircularProgress", () => "circular-progress");

describe("Loader component", () => {
  it("should render correctly", () => {
    const loader = renderer
      .create(
        <MuiThemeProvider theme={lightTheme}>
          <Loader />
        </MuiThemeProvider>
      )
      .toJSON();

    expect(loader).toMatchSnapshot();
  });
});
