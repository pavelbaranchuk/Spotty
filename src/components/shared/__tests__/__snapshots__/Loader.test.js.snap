// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render correctly 1`] = `
<div
  className="Loader-progressContainer-2"
>
  <circular-progress
    className="Loader-progress-1"
    size={160}
  />
</div>
`;
